config.example.js

```diff
@@ -28,5 +28,19 @@ module.exports = {
     requireClosedTasks: true,
     requireGreenBuild: true,
     allowLandWhenAble: true,
+    /** What is provided to a custom rule:
+     *  {
+     *    pullRequest: BB.PullRequest -- see /src/bitbucket/types.d.ts
+     *    buildStatuses: BB.BuildStatus[] -- see /src/bitbucket/types.d.ts
+     *    approvals: string[] -- usernames of all real approvals
+     *    permissionLevel: "read" | "land" | "admin" -- permission level of the user requesting /can-land
+     *  }
+    rules: [
+      {
+        rule: ({ pullRequest, buildStatuses, approvals, permissionLevel }) => true,
+        error: 'This check will always succeed',
```

Comment:
:)

Reply:
We'll want to document this properly at some point.
I know marco wants to start building on top of these already

Reply:
A suggestion here (sorry, should have gotten feedback back to you sooner).

Just for future-proofings sake, could we:
* Make all these checks async (always await on them, I think a synchronous function should Just Work™ too)
* Remove the wrapping object and just have a function that returns either an empty string or an error message.

The second one is a bit weird, in that _normaly_ I'd be all for having these things named explicitly, especially with nice, simple default error messages. But in the future, we might also want to be able to return more complex error messages that might need to be quite dynamic. So I think returning them from the function would make sense? I did think about having it return `true` or an error message, but that gets a bit weird because strings a truthy and although you could detect it, it's just a bit weird...

I dunno, what do you think?